src/python/pybind11/CMakeLists.txt

```diff
@@ -25,13 +28,17 @@ else()
 endif(CMAKE_VERSION VERSION_LESS "3.12")
 
 if (HAS_PYTHON)
-  if (${LINUX_FLAVOR} MATCHES "^(ubuntu|debian)")
-    SET(PKGDIR "dist-packages")
-  elseif (${LINUX_FLAVOR} MATCHES "^(rhel|centos|amzn|fedora|sles)")
-    SET(PKGDIR "site-packages")
-  endif(${LINUX_FLAVOR} MATCHES "^(ubuntu|debian)")
 
-  find_package(pybind11 2.6.0 REQUIRED PATHS "/usr/local/lib/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/${PKGDIR}/pybind11" "/usr/lib/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/${PKGDIR}/pybind11")
+  execute_process(COMMAND bash "-c" "pip3 show pybind11 | grep Location: | cut -d ' ' -f2"
```

Comment:
Should we keep the changes narrowly scoped to WIN32? 
On Linux we have many flavors of distros and the CMake snippets have been tailored to work on all of them. I am a little weary of changing it.

Reply:
That's a fair point, I reverted the linux changes, and kept changes of this PR centralized on windows.

For reference, the reason I made those changes is to account for different build envs, where python may have been installed in a location we weren't anticipating  (e.g. a virtual environments `/usr/local/share/env...` or in WSL my site-packages path was `/home/user/.local/lib/python3.11/site-packages`). I guess this could be a separate PR, verified on more than just Ubuntu/WSL.